src/app/components/navigation-button/navigation-button.component.ts

```diff
@@ -7,5 +19,68 @@ import { Component } from '@angular/core';
     './navigation-button.component.scss',
   ],
 })
-export class NavigationButtonComponent {
+export class NavigationButtonComponent implements OnDestroy {
+  unsubscriber: Subject<boolean> = new Subject<boolean>();
+
+  constructor(
+    private readonly apollo: Apollo,
+    private readonly router: Router
+  ) {}
+
+  openCheckout(): void {
```

Comment:
Extract GraphQL requests into Obeservalbe functions and make one pipe with `concatMap`s and `tap`.
Pseudocode:
```
apolo.query.pipe(
   concatMap(result1 => clearCart(result1.data)),
   concatMap(result2 => addItemsCartMutation(result2.data)),
   tap(result3 => navigate())
)
```